crates/sui-rosetta/src/account.rs

```diff
@@ -44,7 +44,7 @@ pub async fn balance(
             })
     } else {
         let gas_coins = get_coins(&context.state, payload.account_identifier.address).await?;
-        let amount: u64 = gas_coins.iter().map(|coin| coin.amount.value.abs()).sum();
+        let amount: u128 = gas_coins.iter().map(|coin| coin.amount.value.abs()).sum();
```

Comment:
If amounts are `u128`, shouldn't we also check if their sum overflows `u128`?

Reply:
u128 should be enough for the total number of Sui (is it 10B?) + 9 decimal, the sum will never exceed total number of Sui.

Reply:
Good point! for Sui payments we're good!

Reply:
u64 is not enough, in devnet, each genesis coin are close to the u64 max value, merging 2 of them will cause overflow, wondering if we need to change the value of Sui's Coin object from u64 to u128 as well?

example:
https://explorer.devnet.sui.io/objects/0x1fa80479ae228d6d344459741c042fa14efa35a6

Reply:
I'll open an issue to explore this. For this PR we're good.